core/data/src/main/java/uk/gov/gchq/gaffer/data/element/Edge.java

```diff
@@ -33,52 +40,43 @@
  *
  * @see uk.gov.gchq.gaffer.data.element.Edge.Builder
  */
+@JsonDeserialize(builder = Builder.class)
 public class Edge extends Element implements EdgeId {
     private static final Logger LOGGER = LoggerFactory.getLogger(Edge.class);
     private static final long serialVersionUID = -5596452468277807842L;
     private Object source;
     private Object destination;
     private boolean directed;
 
-    Edge() {
-        super();
-    }
-
-    public Edge(final String group) {
-        super(group);
-    }
-
     public Edge(final String group, final Object source, final Object destination, final boolean directed) {
         super(group);
         this.source = source;
         this.destination = destination;
         this.directed = directed;
+        orderVertices();
     }
 
-    public Object getSource() {
-        return source;
+    private Edge(final Builder builder) {
+        super(builder.group);
+        this.source = builder.source;
+        this.destination = builder.destination;
+        this.directed = builder.directed;
+        builder.properties.forEach((n, v) -> this.putProperty(n, v));
```

Comment:
Iterating through properties here since the properties field in Element is private.